### PR TITLE
Output lib array opt

### LINKED ISSUE #557: --output-library does not accept arrays
**Describe the bug**

Unable to pass an array-value to `--output-library`.

```
webpack --output-library-target window --output-library key1 key2 ./src/main.js
webpack --output-library-target window --output-library key1 --output-library key2 ./src/main.js
```

At least one of these formats should pass `["key1", "key2"]` to the `output.library` option. However, only one value can be passed.

**What is the current behavior?**

output is currently something like
```
window.key2=...
```

**Expected behavior**

should be
```
window.key1=window.key1||{},window.key1.key2=...
```

### PR DESCRIPTION
**Pull request by Hoten**
Fixes #557

### CODE REVIEW

**Comment by Hoten** [comment]
I just left for vacation and don't have access to my machine. And to be frank, I find the extra hoop here to make a simple contribution to be cumbersome. Is there a simple way within GitHubs workflow that you could make these sorts of (to people outside of the project maintenance mindset, annoying) changes?

**Comment by ematipico** [comment]
Don't worry, we can rewrite the message of the commit while we do the merge onto develop!

File: bin/convert-argv.js [hunk 1]
```diff
@@ -441,7 +441,10 @@ module.exports = function(...args) {
 
 		ifArg("output-library", function(value) {
 			ensureObject(options, "output");
-			options.output.library = value;
+			if (!options.output.library) {
```
Comment (ev1stensberg) [inline bin/convert-argv.js]: Default value could be an empty array from `yargs`

File: test/binCases/output/output-library-many/stdin.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+"use strict";
+
+module.exports = function testAssertions(code, stdout, stderr) {
+	expect(code).toBe(0);
+	expect(stdout).toEqual(expect.anything());
+	expect(stdout[5]).toContain("bundle.js");
+	expect(stdout[7]).toMatch(/index\.js.*\{0\}/);
+	expect(stderr).toHaveLength(0);
+
+	const output = require("fs").readFileSync(require("path").join(__dirname, "../../../js/bin/output/output-library-many/bundle.js"), "utf-8");
```
Comment (ev1stensberg) [inline test/binCases/output/output-library-many/stdin.js]: import and destruct at top level instead

File: test/binCases/output/output-library-single/test.opts [hunk 3]
```diff
@@ -0,0 +1,5 @@
+./index.js
+-o ../../../js/bin/output/output-library-single/bundle.js
```
Comment (ev1stensberg) [inline test/binCases/output/output-library-single/test.opts]: Don't think this is needed, what's the syntax in the other tests?

File: bin/convert-argv.js (hunk 1 above)
Comment (Hoten) [reply bin/convert-argv.js]: Just tried this, does not work. Default value is not set if a value is given via the CLI - it isn't an initial value. However I did replace this code w/ `ensureArray`

File: test/binCases/output/output-library-many/stdin.js (hunk 2 above)
Comment (Hoten) [reply test/binCases/output/output-library-many/stdin.js]: done

File: test/binCases/output/output-library-single/test.opts (hunk 3 above)
Comment (Hoten) [reply test/binCases/output/output-library-single/test.opts]: removed. only the original output test used this flag, I just copied from there.

**Comment by Hoten** [comment]
Addressed comments, pushed changes + squashed to one commit.

**Comment by Hoten** [comment]
I also wanted a way to quickly run just my tests, so I introduced an env filter for binCases

```
BIN_TEST_CASES_GREP=output npx jest
```

File: test/binCases/output/output-library-single/stdin.js [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+const fs = require("fs");
+const path = require("path");
+
+module.exports = function testAssertions(code, stdout, stderr) {
+	expect(code).toBe(0);
+	expect(stdout).toEqual(expect.anything());
+	expect(stdout[7]).toMatch(/index\.js.*\{0\}/);
+	expect(stderr).toHaveLength(0);
+
+	const output = fs.readFileSync(path.join(__dirname, "../../../js/bin/output/output-library-single/main.js"), "utf-8");
```
Comment (ev1stensberg) [inline test/binCases/output/output-library-single/stdin.js]: this should probably be in an own variable

File: test/binCases/output/output-library-many/stdin.js [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+const fs = require("fs");
+const path = require("path");
+
+module.exports = function testAssertions(code, stdout, stderr) {
+	expect(code).toBe(0);
+	expect(stdout).toEqual(expect.anything());
+	expect(stdout[7]).toMatch(/index\.js.*\{0\}/);
+	expect(stderr).toHaveLength(0);
+
+	const output = fs.readFileSync(path.join(__dirname, "../../../js/bin/output/output-library-many/main.js"), "utf-8");
```
Comment (ev1stensberg) [inline test/binCases/output/output-library-many/stdin.js]: could you abstract this a lil better? Quite long line and slashes that could be trimmed down using `path.resolve`

File: test/binCases/output/output-library-many/stdin.js [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+	expect(output).toContain("window.key1=window.key1||{},window.key1.key2=function");
```
Comment (ev1stensberg) [inline test/binCases/output/output-library-many/stdin.js]: Maybe match this assertion using the array indice and regexes

File: test/binCases/output/output-library-single/stdin.js (hunk 4 above)
Comment (ev1stensberg) [reply test/binCases/output/output-library-single/stdin.js]: See prev comment 👍

File: test/binCases/output/output-library-many/stdin.js [hunk 7]
```diff
@@ -0,0 +1,14 @@
+"use strict";
+
+const fs = require("fs");
```
Comment (ev1stensberg) [inline test/binCases/output/output-library-many/stdin.js]:
```js
const {readFileSync} = require('fs')
const {join} = require('path')


const output = readFileSync(join(__dirname, __dirname, "..", "..", ".., "js", "bin", "output", output-library-many", "main.js"), "utf-8")
```

File: test/binCases/output/output-library-many/stdin.js (hunk 6 above)
Comment (ev1stensberg) [inline test/binCases/output/output-library-many/stdin.js]: `expect(stdout[myIndices]).toMatch(/window.key1=window.key/*\{0\}/);`

File: test/binCases/output/output-library-single/stdin.js (hunk 4 above)
Comment (ev1stensberg) [inline test/binCases/output/output-library-single/stdin.js]: se prev comment

File: test/binCases/output/output-library-single/stdin.js [hunk 8]
```diff
@@ -0,0 +1,14 @@
...
+	expect(output).toContain("window.key1=function");
```
Comment (ev1stensberg) [inline test/binCases/output/output-library-single/stdin.js]: see prev comment

**Comment by ev1stensberg** [comment]
Seems like tests aren't failing. @sokra is this valid?

**Comment by sokra** [comment]
They ain't failing...?

**Comment by ev1stensberg** [comment]
Did so locally on mine, but saw the CI afterwards. Was weird..

**Pull request merged**